### ISSUE TITLE
The angle of rotation is greater than the calculated angle that should be rotated

### ISSUE THREAD

**Issue by qimuyicai** [Scirra/Construct-bugs#7163]
## Problem description

The sprite will rotate 30 degrees each time the user clicks on it. The number of clicks is calculated by the global variable ClickCount. So the angle of rotation should be 30 x ClickCount. I show the result of the sprite's angle and 30 x ClickCount in two texts. If the mouse clicks the sprite many times quickly, the sprite will rotate at an angle greater than 30 x ClickCount.

If the sprite rotates 1 degree every 1/60th of a second, then the sprite's angle will be 1 degree greater than 30xClickCount. If the sprite rotates 2 degrees every 1/60th of a second, then the sprite's angle will be 2 degrees greater than 30xClickCount. And so on.

## Attach a .c3p

[The angle of rotation is greater than the calculated angle that should be rotated.zip](https://github.com/Scirra/Construct-bugs/files/11909566/The.angle.of.rotation.is.greater.than.the.calculated.angle.that.should.be.rotated.zip)

## Steps to reproduce

1. Open the project.
2. Run
3. Clicks the sprite many times quickly
4. The numbers of the two texts will be different

## Observed result

https://github.com/Scirra/Construct-bugs/assets/30488597/99eaa6a7-c8c6-470e-8258-8a0ec2ddb11a

## Expected result
The angle of rotation of the sprite should be the same as the calculated angle

## More details

**Affected browsers/platforms:** 
Chrome, IOS, Android

**Comment by AshleyScirra** [comment]
There appear to be two problems with your event sheet logic here:

1. Angles are cyclical, not linear, and so performing linear maths on them is not always correct. For example it's not true that the angle will be `30 x ClickCount` - it will wrap around after a while.
2. Angles are internally stored in radians and the degrees-to-radians conversion is subject to floating point inaccuracy. According to your events, if the sprite angle is 29.9999999 degrees, it is within 0-30 degrees and so will rotate another 1 degree, leaving it on 30.99999999 degrees, which may end up displayed as 31 degrees. This can explain why it appears to be up to 1 degree off: the logic of events is not accurate enough to deal with circumstances like that.

Neither is a problem with Construct - it's up to your events to handle details like this.

**Comment by qimuyicai** [comment]
Thank you for the reminder, I am enlightened. But I still have a question, why does this only happen after multiple and rapid clicks? For example, if I click every 3 seconds, this problem does not appear.

**Comment by WilsonPercival** [comment]
> Thank you for the reminder, I am enlightened. But I still have a question, why does this only happen after multiple and rapid clicks? For example, if I click every 3 seconds, this problem does not appear.

@qimuyicai Can't repeat. At me at any speed of pressing so.

**Comment by qimuyicai** [comment]
https://github.com/Scirra/Construct-bugs/assets/30488597/9517c744-c968-4d24-ac01-1547c57e755f

I have uploaded a video which is from the c3p file I uploaded. In the video I reproduce this Problem.

A single click is the right number, multiple and consecutive clicks are the wrong number.

The event is simple enough, and I don't think there are any distractions that I can't really figure out. 

Thank you for your time. I do really appreciate your help!
@WilsonPercival @AshleyScirra
